simulation/utils/generate_map.py

```diff
@@ -55,82 +57,101 @@ def get_seed(self):
+        if not len(self.map_points):
+            xs = [point.x for point in self.points_before_interpolation]
+        else:
+            xs = [point[0] for point in self.map_points]
+
         return xs
 
-    def Y(self):
+    
+    def get_Y_list(self):
         """Get list of y coordinates"""
-        ys = [point.y for point in self.points_before_interpolation] if not len(self.map_points) else [point[1] for point in self.map_points]
+        if not len(self.map_points):
+            ys = [point.y for point in self.points_before_interpolation]
+        else:
+            ys = [point[1] for point in self.map_points]
+
         return ys
 
+    
     def __len__(self):
         """ Length of map (points, not segments) """
-        return len(self.points_before_interpolation) if not len(self.map_points) else len(self.map_points)
-
-    # ************
+        if len(self.map_points):
+            return len(self.points_before_interpolation)
+        else:
+            return len(self.map_points)
+        
     def __delitem__(self, index):
         if not len(self.map_points):
             del self.points_before_interpolation[index]
         else:
             del self.map_points[index]
 
-    # ************
-    def is_crossing(self, i, j):
+    def are_points_in_clockwise_order(self, A, B, C):
+        return (C.y - A.y) * (B.x - A.x) < (B.y - A.y) * (C.x - A.x)
+
+    def are_lines_intersecting(self, i, j):
         """Function that checks if two lines ((X[i],Y[i]), (X[i+1], Y[i+1])), ((X[j],Y[j]), (X[j+1],Y[j+1])) intersect"""
         if i == j:
             return False
-        try:
-            return intersect(self.points_before_interpolation[i], self.points_before_interpolation[i + 1],
-                             self.points_before_interpolation[j], self.points_before_interpolation[j + 1])
-        except IndexError:
-            print(f'i = {i}, j = {j}, size = {len(self)}')
-
-    # ************
-    def exterminate_loops(self, step, y_res, ran=40):
+
+        A = self.points_before_interpolation[i]
+        B = self.points_before_interpolation[i + 1]
+        C = self.points_before_interpolation[j]
+        D = self.points_before_interpolation[j + 1]
+        order_1 = self.are_points_in_clockwise_order(A, C, D)
+        order_2 = self.are_points_in_clockwise_order(B, C, D)
+        order_3 = self.are_points_in_clockwise_order(A, B, C)
+        order_4 = self.are_points_in_clockwise_order(A, B, D)
+
+        return order_1 != order_2 and order_3 != order_4
+
+    
+    #STILL NOT CHANGED, WORKING ON IT
+    def delete_map_loops(self, step, angle, points_radius=50):
         """Function that gets rid of most of the loops in map so as to make it a little less crazy"""
         i = 0
         size = len(self.points_before_interpolation)
         while i < size - 1:
             i = size - 2 if i >= size - 1 else i
```

Comment:
w8... czy ten warunek kiedykolwiek może zwrócić True?

Reply:
to jest delete loops, ktorym sie dopiero bede zajmowac. wczesniej chyba wygladal nieco inaczej, jesli mnie pamiec nie myli, ale jak zaczelam go zmieniac to przypomnialam sobie o jednej z naszych ustalonych zasad - wiecej pull requestow, a mniej napchane - wiec wrocilam do poprzedniego stanu (a przynajmniej tak myslalam), zeby zajac sie tym pozniej. W skrócie - na razie ignore, ta linijka i tak nic nie zmienia, a potem jej nie bedzie :)